web/templates/app.tmpl.html

```diff
@@ -13,10 +13,6 @@
 
 {{ define "script-tags" }}
   {{ if not .Contents.Error }}
-    {{ if .Context.WebpackDevServer }}
-      <script type="text/javascript" src="{{.Context.WebpackDevServer}}/dist/index_bundle.js" async></script>
-    {{else}}
-      <script type="text/javascript" src="{{.Contents.PathPrefix}}dist/index_bundle.js" async></script>
-    {{end}}
+    <script type="text/javascript" src="{{.Contents.PathPrefix}}dist/index_bundle.js" async></script>
```

Comment:
😻

Reply:
I think I'm not understanding the setup, but I was expecting the name of this file to change, based on this sentence from the description of this PR:

> Modify the webpack build pipeline to add a hash to the JS bundle filename

Isn't this the JS bundle? I see that you've set cache control headers on this file, so maybe that fixes the issue?

Reply:
This isn't the JS bundle - it's a js file that points to the hashed bundle. so the file that actually contains our js would be the hashed bundle and this file, when generated, would look something like:
```
// based on https://gist.github.com/necolas/1025811
(function(doc, script) {
  let js,
    fjs = doc.getElementsByTagName(script)[0],
    add = function(url, id) {
      if (doc.getElementById(id)) {return;}
      js = doc.createElement(script);
      js.src = url;
      id && (js.id = id);
      fjs.parentNode.insertBefore(js, fjs);
    };

  
    let root = fjs.src.substring(0, fjs.src.indexOf("dist"));
    add(root+'dist/main.17249b45c7d23228c299.js');
  

}(document, 'script'));

```

Reply:
Ah, yep, I was definitely not understanding what was going on here. Thanks for the explanation!

Reply:
Thanks for the explanation @rmars. This conversation reaffirms my thinking that this `app.tmpl.html` -> `index_bundle.js` -> `main.17249b45c7d23228c299.js` dependency chain is overly complex and not very canonical. I think we can roll forward with these changes for now, but I've filed #2074 (thanks @rmars for the idea!) as a way out of this.